### PR TITLE
Finish writing CreateGameServlet.java

### PR DESCRIPTION
**Pull request by ldcWV**
http://ldchen-step-2020.appspot.com/

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/CreateGameServlet.java [hunk 1]
```diff
@@ -40,38 +40,117 @@
...
     }
 
     /**
-    * Checks that the values/sizes of numStages, numHints, stageSpawnLocations, stageStarterHints,
-    * hintLocations, and hintTexts all are consistent with one another.
+    * Creates a Hint object given the stage index and the hint index.
+    * stageIdx = 0 corresponds to the 1st stage, and hintIdx = 0
+    * corresponds to the 1st hint.
+    * @param stageIdx the stage index.
+    * @param hintIdx the hint index within the stage.
```
Comment (saramissak) [inline src/main/java/com/google/sps/servlets/CreateGameServlet.java]: "Idx" is unclear what it is, try to just put index to make it more clear that it is the index.

File: src/main/java/com/google/sps/servlets/CreateGameServlet.java [hunk 2]
```diff
@@ -40,38 +40,117 @@
...
+
+    /**
+    * Creates a Stage object given the stage index.
+    * stageIdx = 0 corresponds to the 1st stage.
+    * @param stageIdx the stage index.
+    * @return a Stage object corresponding to the stage index.
+    */
+    Stage buildStage(int idx) {
+        String stageID = IDGenerator.gen();
+        Stage.Builder res = new Stage.Builder(stageID, idx+1);
```
Comment (saramissak) [inline src/main/java/com/google/sps/servlets/CreateGameServlet.java]: Why is there a +1 here?

File: src/main/java/com/google/sps/servlets/CreateGameServlet.java (hunk 1 above)
Comment (ldcWV) [reply src/main/java/com/google/sps/servlets/CreateGameServlet.java]: Changed, thanks!

File: src/main/java/com/google/sps/servlets/CreateGameServlet.java (hunk 2 above)
Comment (ldcWV) [reply src/main/java/com/google/sps/servlets/CreateGameServlet.java]: The stage numbers are 1 indexed, but the indices in the array are 0 indexed.

**Review by saramissak — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/CreateGameServlet.java (hunk 2 above)
Comment (saramissak) [reply src/main/java/com/google/sps/servlets/CreateGameServlet.java]: Okay makes sense!

**Pull request merged**